elastiknn-plugin/src/main/scala/com/klibisz/elastiknn/mapper/VectorMapper.scala

```diff
@@ -179,14 +127,9 @@ abstract class VectorMapper[V <: Vec: ElasticsearchCodec] { self =>
 
         override def doXContentBody(builder: XContentBuilder, params: ToXContent.Params): Unit = {
           super.doXContentBody(builder, params)
-          ElasticsearchCodec
-            .encode(mapping)
-            .asObject
-            .map(_.toIterable.filter(_._1 != "type"))
-            .map(JsonObject.fromIterable)
-            .map(Json.fromJsonObject)
-            .foreach(populateXContent(_, builder))
-
+          // Note: encoding method should NOT write the "type" property to the builder.
```

Comment:
This is the reason for the MappingEncoder. Elasticsearch only wants the mapping internals without the type. It wants `"elastiknn": { ... }` instead of `{"type":"...","elastiknn":{...}}`.